clients/drcachesim/common/trace_entry.h

```diff
@@ -409,8 +409,12 @@ typedef enum {
     TRACE_MARKER_TYPE_PAGE_SIZE,
 
     /**
-     * This marker is emitted prior to each system call when -enable_kernel_tracing is
-     * specified. The marker value contains a unique system call identifier.
+     * This marker is emitted after each system call that records kernel PT, and its value
```

Comment:
See the comment I added to event_post_syscall:

    /* TODO i#5505: We should move this _IDX marker to pre-syscall for two
     * reasons: 1) Some syscalls have no post event (e.g., exit or sigreturn);
     * 2) A blocking syscall may have a thread switch in between, separating the
     * syscall instruction from this marker by quite a distance when interleaved
     * with other threads.
     */

What was the original reason to place the marker in post-syscall?  Is there any advantage to doing so?  What about these significant disadvantages in the comment?

Reply:
Done.